File: CodeAnalysis Test Cases/TestCasesCS/Quality/MagicNumbersInConditions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSTestCases.Quality
{
    public class MagicNumbersInConditions
    {
        int x = 8;

        public bool IsGood(string password)
        {
            if (password.Length < 5)
                return false;

            return password.Length >= 7;
        }

        public int fun()
        {
            int[] g = null;

            return g[x];
        }

        public bool zun()
        {
            float z = 0.0F;

            if (z > 3.4)
                return false;
            else
                return true;
        }
    }
}

File: CodeAnalysis Test Cases/TestCasesVB/Quality/MagicNumbersInConditions.vb
Public Class MagicNumbersInConditions
    Private x As Integer = 8

    Public Function IsGood(password As String) As Boolean
        Dim flag As Boolean = password.Length < 5
        Return Not flag AndAlso password.Length >= 7
    End Function

    Public Function fun() As Integer
        Dim g As Integer() = Nothing
        Return g(Me.x)
    End Function

    Public Function zun() As Boolean
        Dim z As Single = 0F
        Dim flag As Boolean = CDec(z) > 3.4
        Return Not flag
    End Function
End Class
